src/commands/synthetics/test.ts

```diff
@@ -31,18 +30,17 @@ export const getTestConfigs = async (
 
 export const getTestsFromSearchQuery = async (
   api: APIHelper,
-  config: RunTestsCommandConfig,
-  reporter: MainReporter
+  config: Required<Pick<RunTestsCommandConfig, 'global' | 'testSearchQuery'>>
 ) => {
-  const testsToTriggerBySearchQuery = await getTestListBySearchQuery(api, config.global, config.testSearchQuery || '')
+  const {global: globalConfigOverride, testSearchQuery} = config
```

Comment:
nit: we could move the test on `testSearchQuery` inside this function to avoid this slightly more complex type here, and the need for the user-defined guard `hasTestSearchQuery`.

```suggestion
  if (!config.testSearchQuery) {
    return []
  }
  
  const {global: globalConfigOverride, testSearchQuery} = config
```

Reply:
Very true